torchx/c_src/torchx.cpp

```diff
@@ -351,6 +351,19 @@ NIF(permute)
   TENSOR(t->permute(dims).contiguous());
 }
 
+NIF(take)
+{
+  TENSOR_PARAM(0, t);
+  TENSOR_PARAM(1, i);
+  PARAM(2, int64_t, axis);
+
+  if (axis == 0)
+  {
+    TENSOR(t->index({*i}));
+  } else {
+    TENSOR(t->transpose(0, axis).index({*i}).transpose(0, axis));
+  }
```

Comment:
@josevalim I started writing this code, but ended up ditching it because I would have to basically rewrite the code which I moved into `Nx.Shape` here to make things work for axis != 0.

We have some options:

1. Ditch this C++ code altogether in favor of using `gather` (which is the current code, minus erasing this file)
2. Call this C++ version only for axis = 0 as a kind of optimization
3. Rewrite the "fully-qualified" code here + moving the "index offsets" thing which we use in gather to C++ so we can use here as well.

Reply:
Generally speaking, I would prefer to keep the C++ code to a thin wrapper around Libtorch API. So if Libtorch does not expose a `take`, I wouldn't expose one here either.

The only exception is if we can use several destructive operations (which we don't expose to Elixir for obvious reasons).

Reply:
I see! Perhaps I should remove take_along_axis then, in a future PR?

Reply:
Yeah, you probably want to rename Torchx.take_along_axis to Torchx.gather and then implement the backend gather using Torchx APIs. :)

Reply:
In general, our Torchx module should really mimic Libtorch actual functions. :)